### PR TITLE
Added title to be displayed for stackexchange question

### LINKED ISSUE #16225: Missing title in Stack Overflow Embed
**Describe the bug**

I noticed that the Stack Overflow liquid tag is missing a title on this post:

https://dev.to/twitterdev/getting-the-follower-count-from-twitter-api-v2-4jh6

**To Reproduce**

- Go to `/new` locally
- Paste this and preview `{% stackoverflow 70779246 %}`

**Expected behavior**

The title shows up

**Screenshots**

Screen Shot 2022-01-20 at 10 21 33 AM

**Additional context**
This won't be "fixed" in prod until the post is re-saved.

### PR DESCRIPTION
**Pull request by sabarish98**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
-  [x] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description
Title of questions don't get displayed while StackOverflow / StackExchange Liquid Tags are used. This change fixes that.
## Related Tickets & Documents
Resolves #16225 
## QA Instructions, Screenshots, Recordings

To Reproduce:

1. Go to New Article Page (/new)
2. Paste this and preview {% stackoverflow 70779246 %}

OR refer the issue description for quick reference.

### UI accessibility concerns?

_If your PR includes UI changes, please replace this line with details on how
accessibility is impacted and tested. For more info, check out the
[Forem Accessibility Docs](https://developers.forem.com/frontend/accessibility)._

UI Accessibility Testing won't be needed since there is no impact

## Added/updated tests?

- [x] Yes
- [ ] No
- [ ] I need help with writing tests

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

- [ ] I've updated the [Developer Docs](https://developers.forem.com) or
      [Storybook](https://storybook.forem.com/) (for Crayons components)
- [ ] This PR changes the Forem platform and our documentation needs to be
      updated. I have filled out the
      [Changes Requested](https://github.com/forem/admin-docs/issues/new?assignees=&labels=&template=changes_requested.md)
      issue template so Community Success can help update the Admin Docs
      appropriately.
- [ ] I've updated the README or added inline documentation
- [ ] I've added an entry to
      [`CHANGELOG.md`](https://github.com/forem/forem/tree/main/CHANGELOG.md)
- [ ] I will share this change in a [Changelog](https://forem.dev/t/changelog)
      or in a [forem.dev](http://forem.dev) post
- [ ] I will share this change internally with the appropriate teams
- [ ] I'm not sure how best to communicate this change and need help
- [ ] This change does not need to be communicated, and this is why not: _please
      replace this line with details on why this change doesn't need to be
      shared_

## [optional] What gif best describes this PR or how it makes you feel?

alt_text

### CODE REVIEW

**Comment by citizen428** [comment]
Can you please look into the spec failure on Travis, it seems to legitimately fail for this spec.

**Review by msarit — COMMENTED** [review]
@sabarish98 Thank you for this PR, I'll complete my review shortly.

@djuber I'm pinging you on this PR cos those superfluous comments are showing up in my local (again, see pic below 🤦🏾‍♀️ ). Is this happening for you also?

Screen Shot 2022-01-24 at 2 57 27 PM

**Review by msarit — APPROVED** [review]

**Comment by djuber** [comment]
@msarit - yes - I also get the code block with the </p> tag (it's the same reason as for other liquid tags, in development the view is rendering an html comment with the erb template filename, from `ActionView::Base.annotate_rendered_view_with_filenames`, and the markdown parser is getting confused by a closing tag, followed by a four space indent, and interpreting the `</p>` as a code block).

That's a non-blocker (annotation is off in production, we should resolve this independently of this PR).

Other than that, works as expected, given this input:

```
{% stackoverflow 70779246 %}
```

I get the same resulting post as the image you showed.

**Review by djuber — APPROVED** [review]
one test cleanup/deduplication suggestion, otherwise looks good, thank you.

File: spec/liquid_tags/stackexchange_tag_spec.rb [hunk 1]
```diff
@@ -26,6 +26,16 @@ def generate_exchange_liquid(id)
       end
     end
 
+    it "renders basic stackoverflow html with title" do
+      VCR.use_cassette("stackexchange_tag_stackoverflow") do
+        liquid = generate_new_liquid(valid_id)
+        expect(liquid.render).to include("ltag__stackexchange")
+        expected_answer_url = "https://stackoverflow.com/questions/57496168/use-where-and-limit-to-child-in-foreach"
+        expect(liquid.render).to include("<a href=\"#{expected_answer_url}\"")
+        expect(liquid.render).to include("Use &quot;where&quot; and &quot;limit&quot; to child in @foreach")
```
[the remark this answers is not in the document]
Comment (sabarish98) [reply spec/liquid_tags/stackexchange_tag_spec.rb:35]: That's a valid suggestion @djuber. I will incorporate this via a new commit. I shall add a separate case for the title fix

[the remark this answers is not in the document]
Comment (sabarish98) [reply spec/liquid_tags/stackexchange_tag_spec.rb:32]: I have added the case now : )

**Comment by djuber** [comment]
I'm going to merge this and follow up with a test case cleanup - no need to wait on that tiny point.

**Pull request merged**